pkg/vault/contracts/Vault.sol

```diff
@@ -647,6 +648,7 @@ contract Vault is IVaultMain, VaultCommon, Proxy {
             poolData.poolConfigBits.requireUnbalancedLiquidityEnabled();
 
             amountsInScaled18 = maxAmountsInScaled18;
+            amountsInRaw = params.maxAmountsIn;
```

Comment:
Note that here we're discarding the original array and pointing to `maxAmountsIn`.

In this case `maxAmountsIn` should not be modified. It might be better to just make `amountsInRaw` a copy of `maxAmountsIn` instead.

Reply:
I've added a simple deep-copy function for arrays.

The compiler doesn't seem to be optimizing the loop, even in 0.8.26. `mcopy` could be an alternative to save some bytes, but I don't want to optimize early.